_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_slice.cpp

```diff
@@ -18,38 +18,39 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-//  MPEG-2 is a international standard promoted by ISO/IEC and
-//  other organizations. Implementations of this standard, or the standard
-//  enabled platforms may require licenses from various entities, including
-//  Intel Corporation.
-
-#pragma once
-
 #include "umc_defs.h"
 #if defined (MFX_ENABLE_MPEG2_VIDEO_DECODE)
 
-#include "umc_video_decoder.h"
+#include "umc_mpeg2_defs.h"
+#include "umc_mpeg2_slice.h"
 
-//#define OTLAD
-
-namespace UMC
+namespace UMC_MPEG2_DECODER
 {
-
-// profile and level definitions for mpeg2 video
-// values for profile and level fields in BaseCodecParams
-#define MPEG2_PROFILE_SIMPLE  5
-#define MPEG2_PROFILE_MAIN    4
-#define MPEG2_PROFILE_SNR     3
-#define MPEG2_PROFILE_SPATIAL 2
-#define MPEG2_PROFILE_HIGH    1
-
-#define MPEG2_LEVEL_LOW   10
-#define MPEG2_LEVEL_MAIN   8
-#define MPEG2_LEVEL_H14    6
-#define MPEG2_LEVEL_HIGH   4
-
-    class MPEG2VideoDecoderBase;
-    
+    bool MPEG2Slice::Reset()
+    {
+        m_bitStream.Reset((uint8_t *) source.GetDataPointer(), (uint32_t) source.GetDataSize());
+
+        if (source.GetDataSize() && false == DecodeSliceHeader())
```

Comment:
If source.GetDataSize() ==0 the method returns true. Is it OK?

Reply:
changed